tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+from nailgun import client
+from random import randint, shuffle
+from requests.exceptions import HTTPError
+from robottelo import entities
+from robottelo.common.constants import DOCKER_REGISTRY_HUB
+from robottelo.common.decorators import (
+    data, run_only_on, skip_if_bug_open, stubbed)
+from robottelo.common.helpers import get_server_credentials
+from robottelo.test import APITestCase
+# (too-many-public-methods) pylint:disable=R0904
+
+
+STRING_TYPES = ['alpha', 'alphanumeric', 'cjk', 'utf8', 'latin1']
+
+
+def _create_repository(prod_id, name=None, upstream_name=None):
+    """Creates a Docker-based repository.
+
+    :param str name: Name for the repository. If ``None`` then a random
+    value will be generated.
+    :param str upstream_name: A valid name for an existing Docker image.
+    If ``None`` then defaults to ``busybox``.
+
+    :return: A dictionary representing the created repository.
+
+    """
+
+    if name is None:
+        name = gen_string(gen_choice(STRING_TYPES), 15)
+    if upstream_name is None:
+        upstream_name = u'busybox'
+
+    return entities.Repository(
+        product=prod_id,
+        content_type=u'docker',
+        name=name,
+        docker_upstream_name=u'busybox',
+        url=DOCKER_REGISTRY_HUB
+    ).create()
+
+
+def _create_content_view(org_id, composite=False):
+    """Creates a contenr view.
+
+    :param id org_id: The ID for an existing organization.
+    :param bool composite: Whether the content view is composite or not.
+    Defaults to ``False``.
+
+    :return: A dictionary representing the created content view.
+
+    """
+    return entities.ContentView(
+        organization=org_id, composite=composite).create()
+
+
+def _add_repo_to_content_view(repo_id, cv_id):
+    """Adds a repository to an existing content view.
+
+    :param int repo_id: The ID for an existing repository.
+    :param int cv_id: The ID for an existing content view.
+
+    """
+    client.put(
+        entities.ContentView(id=cv_id).path(),
+        auth=get_server_credentials(),
+        verify=False,
+        data={u'repository_ids': [repo_id]}
+    ).raise_for_status()
+
+
+def _add_content_view_to_composite_view(cv_id, cv_version_id):
+    """Adds a published content view to a composite content view.
+
+    :param int cv_id: The ID for an existing composite content view.
+    :param int cv_version_id: The ID for a published non-composite
+    content view.
+
+    """
+    client.put(
+        entities.ContentView(id=cv_id).path(),
+        auth=get_server_credentials(),
+        verify=False,
+        data={u'component_ids': [cv_version_id]}
+    ).raise_for_status()
+
+
+def _publish_content_view(cv_id):
+    """Publishes an existing content view.
+
+    :param int cv_id: The ID for an existing content view.
+
+    :return: A boolean indicating whether the publication worked or not.
+
+    """
+    task_status = entities.ContentView(id=cv_id).publish()
+
+    return task_status['result'] == u'success'
```

Comment:
I'm seeing a lot of code like this sprinkled throughout the code base. Perhaps methods like like `publish` and `promote` should be modified to check for task success. Thoughts?